### ISSUE TITLE
Error accessing iOS 11 constant string definition

### ISSUE THREAD

**Issue by erjdriver** [NativeScript/NativeScript#5029]
I'm trying to access iOS 11 specific definition UITextContentTypeUsername (string).

I've got tns-platform-declarations 3.3

NS 3.3.

The TS code compiles fine but when I run it on an IOS 11 device:

`JS ERROR ReferenceError: Can't find variable: UITextContentTypeUsername`

I get the error even if a refer to it to check if it's undefined.

`if ( UITextContentTypeUsername != undefined )...`

The variable is defined in IOS platform declarations in UIKit file of tns-platform-declarations

I tried another textContentType UITextContentTypeName (defined in IOS 10) and that works fine - at least no crash.

Here's how I'm using it on my ts file:

```
let field = page.getViewById<textField.TextField>("field");

 field = UITextContentTypeName; // ok

 field = UITextContentTypeUsername; // fails
```

Again this test was done on IOS 11 device.

**Comment by tsonevn** [comment]
Hi @erjdriver,
I tested your case while using the latest version of `tns-platform-declarations` 3.3, However, I was unable to recreate the issue with accessing `UITextContentTypeUsername` in iOS 11. For your convenience, I am attaching GIF file from the simulator and sample project. Please review the project and make the needed changes, which will allow us to recreate this behavior.

In the meantime, you could you try deleting `node_modules` and `platforms` folders and to rebuild the application.

[Archive.zip](https://github.com/NativeScript/NativeScript/files/1440477/Archive.zip)
![screencast 2017-11-03 at 10 52 41 am](https://user-images.githubusercontent.com/17448734/32366296-2e84d5e2-c086-11e7-9aa8-5c3df1b5ac78.gif)

**Comment by erjdriver** [comment]
Thank you.

I downloaded your archive file and ran it (new node_modules and platform).

I tried it on a pre iOS 11 emulator and it crashes as soon as you reference the variable. I believe it's a string constant and should be there for any IOS version - right?

Note that the output prints your first console.log string and then crashes when it references the constant string.

Again this is on a pre iOS 11 emulator.


`Installing...
Successfully installed on device with identifier '88691FE0-31E2-4F2F-AD28-4023D1597C29'.
Successfully transferred all files.
Refreshing application...
Successfully synced application org.nativescript.i5029 on device 88691FE0-31E2-4F2F-AD28-4023D1597C29.
Executing before-watch hook from /Users/asrock/Documents/test9/hooks/before-watch/nativescript-dev-typescript.js
Found peer TypeScript 2.4.2
Class PLBuildVersion is implemented in both <snipped> One of the two will be used. Which one is undefined.
CONSOLE LOG file:///app/main-page.js:32:16: ----------UITextContentTypeUsername-----------
1   0x10f6b5d4b NativeScript::FFICallback<NativeScript::ObjCMethodCallback>::ffiClosureCallback(ffi_cif*, void*, void**, void*)
2   0x10fd5928e ffi_closure_unix64_inner
3   0x10fd59bd2 ffi_closure_unix64
4   0x110a80073 -[UIViewController __viewWillAppear:]
5   0x110ab9801 -[UINavigationController _startTransition:fromViewController:toViewController:]
6   0x110aba6b3 -[UINavigationController _startDeferredTransitionIfNeeded:]
7   0x110abb7f1 -[UINavigationController __viewWillLayoutSubviews]
8   0x110cad2bc -[UILayoutContainerView layoutSubviews]
9   0x11099a20b -[UIView(CALayerDelegate) layoutSublayersOfLayer:]
10  0x115297904 -[CALayer layoutSublayers]
11  0x11528b526 CA::Layer::layout_if_needed(CA::Transaction*)
12  0x11528b3a0 CA::Layer::layout_and_display_if_needed(CA::Transaction*)
13  0x11521ae92 CA::Context::commit_transaction(CA::Transaction*)
14  0x115247130 CA::Transaction::commit()
15  0x115247b37 CA::Transaction::observer_callback(__CFRunLoopObserver*, unsigned long, void*)
16  0x11370f717 __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__
17  0x11370f687 __CFRunLoopDoObservers
18  0x1136f4038 CFRunLoopRunSpecific
19  0x1108d102f -[UIApplication _run]
20  0x1108d70d4 UIApplicationMain
21  0x10fd59a2d ffi_call_unix64
22  0x12c5f9250
file:///app/main-page.js:33:42: JS ERROR ReferenceError: Can't find variable: UITextContentTypeUsername
7:24:20 AM - Compilation complete. Watching for file changes.
`
